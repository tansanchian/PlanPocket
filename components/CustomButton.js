import {
  View,
  Text,
  StyleSheet,
  Pressable,
  Animated,
  TouchableOpacity,
} from "react-native";
import React from "react";

const CustomButton = ({
  onPress,
  text,
  design = "MAIN",
  type = "PRIMARY",
  bgColor,
  fgColor,
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        styles[`container_${design}`],
        styles[`container_${type}`],
        bgColor ? { backgroundColor: bgColor } : {},
      ]}
    >
      <Text
        style={[
          styles.text,
          styles[`text_${type}`],
          fgColor ? { color: fgColor } : {},
        ]}
      >
        {text}
      </Text>
    </TouchableOpacity>
  );
};

export default CustomButton;

const styles = StyleSheet.create({
  container_MAIN: {
    paddingVertical: 15,
    borderRadius: 25,
    marginVertical: 10,
    width: "100%",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 5,
  },
  container_HALF: {
    paddingVertical: 15,
    borderRadius: 25,
    marginVertical: 10,
    width: "50%",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 5,
  },
  container_RIGHT: {
    alignItems: "flex-end",
  },
  text_RIGHT: {
    color: "grey",
  },
  text: {
    fontWeight: "bold",
    color: "white",
  },

  container_PRIMARY: {
    backgroundColor: "#735DA5",
  },

  container_TERTIARY: {},

  text_TERTIARY: {
    color: "grey",
  },
});
