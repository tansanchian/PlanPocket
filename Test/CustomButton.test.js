import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import CustomButton from "../components/CustomButton";

describe("CustomButton", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<CustomButton text="Test" onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("calls onPress function when the button is pressed", () => {
    const mockOnPress = jest.fn();

    const { getByTestId } = render(
      <CustomButton text="Test" onPress={mockOnPress} />
    );

    const pressMeButton = getByTestId("MyButton:Button:ClickMe");
    fireEvent.press(pressMeButton);

    expect(mockOnPress).toHaveBeenCalled();
  });
});
