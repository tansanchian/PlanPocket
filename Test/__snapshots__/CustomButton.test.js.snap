// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly and matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#735DA5",
      "borderRadius": 25,
      "marginVertical": 10,
      "opacity": 1,
      "paddingVertical": 15,
      "width": "100%",
    }
  }
  testID="MyButton:Button:ClickMe"
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontWeight": "bold",
        },
        undefined,
        {},
      ]
    }
  >
    Test
  </Text>
</View>
`;
